src/content/contribute/writing-a-plugin.md

```diff
@@ -133,76 +154,106 @@ Once we can latch onto the webpack compiler and each individual compilations, th
 Let's write a simple example plugin that generates a new build file called `filelist.md`; the contents of which will list all of the asset files in our build. This plugin might look something like this:
 
 ```javascript
-function FileListPlugin(options) {}
-
-FileListPlugin.prototype.apply = function(compiler) {
-  compiler.plugin('emit', function(compilation, callback) {
-    // Create a header string for the generated file:
-    var filelist = 'In this build:\n\n';
-
-    // Loop through all compiled assets,
-    // adding a new line item for each filename.
-    for (var filename in compilation.assets) {
-      filelist += ('- '+ filename +'\n');
-    }
-
-    // Insert this list into the webpack build as a new file asset:
-    compilation.assets['filelist.md'] = {
-      source: function() {
-        return filelist;
-      },
-      size: function() {
-        return filelist.length;
+class FileListPlugin {
+  apply(compiler) {
+    // emit is asynchronous hook, tapping into it using tapAsync, you can use tapPromise/tap(synchronous) as well
+    compiler.hooks.emit.tapAsync("FileListPlugin", (compiler, callback) => {
+      // Create a header string for the generated file:
+      var filelist = "In this build:\n\n";
+
+      // Loop through all compiled assets,
+      // adding a new line item for each filename.
+      for (var filename in compilation.assets) {
+        filelist += "- " + filename + "\n";
       }
-    };
 
-    callback();
-  });
-};
+      // Insert this list into the webpack build as a new file asset:
+      compilation.assets["filelist.md"] = {
+        source: function() {
+          return filelist;
+        },
+        size: function() {
+          return filelist.length;
+        }
+      };
+
+      callback();
+    });
+  }
+}
 
 module.exports = FileListPlugin;
 ```
 
 ## Different Plugin Shapes
 
-A plugin can be classified into types based on the event it is registered to. Every event hook decides how it is going to apply the plugins in its registry.
+A plugin can be classified into types based on the event hooks it taps into. Every event hook is pre-defined as synchronous/asynchronous/waterfall/parallel and hook is called internally using call/callAsync method. The list of hooks that are supported/can be tapped into are generally specified in this.hooks property.
```

Comment:
supported or can be tapped...

I would recommend to use or and comma separation here for the  `/` synchronous/asynchronous/waterfall/parallel